### DISCUSSION TITLE
Checking redundant and satisfied constraints.

### DISCUSSION THREAD

**Discussion by martinbb0** [casadi/casadi#3592] [Q&A] [answered]
Hi! Is there an easy way to identify which constraints of a problem are fully satisfied? By satisfied, I mean that if, for instance, a length has a maximum constraint, the optimal solution would reach that maximum value.

**Comment by brunoml5** [comment, marked as answer]
Hi! The [Opti](https://web.casadi.org/docs/#document-opti) interface has some cool functions for debugging.

You can save the constraint expression and then probe the value after solving, just like the example in the documentation.
```
con = sin(x+y)>=1
opti.subject_to(con)
sol = opti.solve()
sol.value(con)
```
You can also check the values for all expressions at once.
```
print(opti.g)
sol.value(opti.g)
```
Or the dual variables with:
```
con = sin(x+y)>=1;
opti.subject_to(con);
sol = opti.solve();

sol.value(opti.dual(con))
```
Or `sol.value(opti.lam_g)`

PS: Don't forget that the values most likely won't be zero for the constraints because most methods satisfy them according to an error tolerance.
